trieste/models/gpflux/sampler.py

```diff
@@ -331,42 +338,79 @@ def _prepare_weight_sampler(self) -> Callable[[int], TensorType]:  # [B] -> [B,
         canonical features for `P` outputs.
         """
 
-        if isinstance(self._layer.inducing_variable, InducingPoints):
+        if isinstance(self._layer.inducing_variable, SharedIndependentInducingVariables):
             inducing_points = self._layer.inducing_variable.Z  # [M, D]
+            separate_case = False
```

Comment:
perhaps you can set this as a class attribute when checking `if isinstance(layer.kernel, gpflow.kernels.SharedIndependent)` and then reuse it throughout?

Reply:
True